tests/foreman/ui/test_computeresource_ec2.py

```diff
@@ -0,0 +1,382 @@
+               image from and submit.
+
+        :expectedresults: The image should not be added to the CR
+
+        :Caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier2
+    def test_positive_access_ec2_with_default_profile(self):
+        """Associate default (3-Large) compute profile to ec2 compute
+        resource
+
+        :id: 11577087-8c0c-4e87-aed0-4d1b147fd274
+
+        :setup: ec2 hostname, credentials, and flavor.
+
+        :steps:
+
+            1. Create a compute resource of type ec2.
+            2. Provide a valid Access Key and Secret Key.
+            3. Select the created ec2 CR.
+            4. Click Compute Profile tab.
+            5. Select (3-Large) and submit.
+
+        :expectedresults: The compute resource created and opened successfully
+
+        :Caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier2
+    def test_positive_access_ec2_with_custom_profile(self):
+        """Associate custom (3-Large) compute profile to ec2 compute resource
+
+        :id: 88cb2f19-4f6e-4533-859b-59c7d99c206f
+
+        :setup: ec2 hostname and credentials, custom flavor.
+
+        :steps:
+
+            1. Create a compute resource of type ec2.
+            2. Provide a valid Access Key and Secret Key.
+            3. Select the created ec2 CR.
+            4. Click Compute Profile tab.
+            5. Edit (3-Large) with valid configurations and submit.
+
+        :expectedresults: The compute resource created and opened successfully
+
+        :Caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier2
+    def test_positive_retrieve_ec2_vm_list(self):
+        """List the virtual machines from ec2 compute resource.
+
+        :id: a8f09878-4aec-484c-a9d6-710ffe2cc8df
+
+        :setup: ec2 hostname and credentials.
+
+        :steps:
+
+            1. Select the created compute resource.
+            2. Go to "Virtual Machines" tab.
+
+        :expectedresults: The Virtual machines should be displayed
+
+        :Caseautomation: notautomated
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_provision_ec2_host_with_image(self):
+        """ Provision a host on ec2 compute resource with image based
+
+        :id: 5b68bfb8-6c15-4629-b25a-8928afdc7ca9
+
+        :setup: ec2 hostname and credentials.
+            1. Addition of custom provisioning template
+               to add ssh-key parameter.
+
+        :steps:
+            1. Populate images into satellite from ec2 images.
+            2. Associate Activation key and CV to the host.
+            3. Edit the required Operating system to use the custom
+               provisioning template.
+            4. Go to "Hosts --> New host".
+            5. Fill in the required details.(eg name,loc, org)(hostgroup)
+            6. Select ec2 compute resource from "Deploy on" drop down.
```

Comment:
I feel 3 will be part of the test-case and hence, as we have to ensure it exists for every host, so that we can login, with our own keys, as foreman uses it's own keys during configuring the instance.